### PR TITLE
Additional Composer script prerequisite

### PR DESCRIPTION
**Pull request by raphaelstolt**
Allows to run a set of [Composer scripts](https://getcomposer.org/doc/articles/scripts.md).

### CODE REVIEW

**Comment by ppetermann** [comment]
can you elaborate on which purpose the composer scripts during the RMT run serve?

**Comment by raphaelstolt** [comment]
Each Composer script has it's own task specific purpose and is generally a more modern approach in comparsion to direct / specific command runs e.g. `phpunit --stop-on-failure`.

You might have one e.g. `composer test` to run unit and integration tests with a abstraction or wrapper for the testing framework of choice (i.e. PHPUnit, PHPSpec, Codeception, Behat). You might have one to run fix PHP coding standard violations e.g. `composer cs-check|fix` via PHP CodeSniffer or the PHP Coding Standards Fixer, or one to validate commit messages. 

Summarized: the tool specific commands and complexity is hidden, but also generalized, behind a Composer script which tools like RMT would benefit from.

This PR would enable projects to reuse their existing Composer scripts having release relevant purposes.

**Comment by ppetermann** [comment]
hum? composer scripts are usually hooks that are called during certain things composer does, so you can have it shuffle files around, or clean up a skeleton when doing a create-project.

the only way to do something like "composer cs-check" would be a custom command, and that seems to serve no purpose, as there is no benefit of running composer just so it is run through composer, especially as the suggested task you are saying is completely unrelated to managing the external dependencies?

tasks like the ones you are suggesting should be done by RMT directly, without invoking the composer runtime, that actually has nothing to do with the task.

File: test/Liip/RMT/Tests/Unit/Prerequisite/ComposerScriptTest.php [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+     */
+    public function testExceptionWhenNoScriptsProvided()
+    {
+        $options = array(
+            'composer' => 'composer'
+        );
+
+        new ComposerScriptCheck($options);
+    }
+}
```
Comment (jeanmonod) [inline test/Liip/RMT/Tests/Unit/Prerequisite/ComposerScriptTest.php]: Missing a test when the script is working. Have you tried to write one?

**Comment by jeanmonod** [comment]
I'm following @ppetermann on this one. I don't see the added value of calling composer script at release time. But if it could be useful for you, why not. The only thing I would like to see before merging this is a functional test that ensure this extension is working. @raphaelstolt Would you have time to do it?

**Comment by raphaelstolt** [comment]
@jeanmonod Will try to add a functional test over the next few days; if I don't get around doing so feel free to close this PR.

**Comment by jeanmonod** [comment]
Ok, thanks in advance for the func test

**Pull request closed without merging**